### PR TITLE
refactor(client): Replace common-utils with core-utils and client-utils

### PR DESCRIPTION
**Pull request by tylerbutler**
In #16856 we copied some functions out of common-utils. This PR updates the client release group to use the new client-only packages instead of common-utils. **common-utils is no longer used in the client release group.**

To make it easier to review such a large PR, I have split the changes by directory into individual commits. To review just one section of the codebase, select the commit from the drop down in the GitHub UI.

image

Or use these links:

- [client - packages/dds](https://github.com/microsoft/FluidFramework/pull/16871/commits/0faa3b2d894da4e4670ec099711ac32deca6be39)
- [client - packages/drivers](https://github.com/microsoft/FluidFramework/pull/16871/commits/4775bcb81b)
- [client - packages/framework](https://github.com/microsoft/FluidFramework/pull/16871/commits/8a17672be8)
- [client - packages/loader](https://github.com/microsoft/FluidFramework/pull/16871/commits/737d98ab13)
- [client - packages/runtime](https://github.com/microsoft/FluidFramework/pull/16871/commits/6849cec3fd)
- [client - packages/test](https://github.com/microsoft/FluidFramework/pull/16871/commits/226a4fa2f4)
- [client - packages/tools](https://github.com/microsoft/FluidFramework/pull/16871/commits/48eb200866)
- [client - packages/utils](https://github.com/microsoft/FluidFramework/pull/16871/commits/67dabb32e3)
- [client - azure](https://github.com/microsoft/FluidFramework/pull/16871/commits/aa278b51ea)
- [client - examples](https://github.com/microsoft/FluidFramework/pull/16871/commits/f8d21969ca)
- [client - experimental](https://github.com/microsoft/FluidFramework/pull/16871/commits/b23e3cdaff)
- [client - config and doc changes](https://github.com/microsoft/FluidFramework/pull/16871/commits/0650a5b7d0)

### CODE REVIEW

File: examples/version-migration/schema-upgrade/package.json [hunk 1]
```diff
@@ -38,13 +38,14 @@
...
 		"@fluid-experimental/react-inputs": "workspace:~",
+		"@fluid-internal/client-utils": "workspace:~",
 		"@fluidframework/aqueduct": "workspace:~",
 		"@fluidframework/cell": "workspace:~",
-		"@fluidframework/common-utils": "^1.1.1",
 		"@fluidframework/container-definitions": "workspace:~",
 		"@fluidframework/container-loader": "workspace:~",
 		"@fluidframework/container-runtime-definitions": "workspace:~",
 		"@fluidframework/core-interfaces": "workspace:~",
+		"@fluidframework/core-utils": "workspace:~",
```
Comment (alexvy86) [inline examples/version-migration/schema-upgrade/package.json:48]: I think I see several instances of packages where the common-utils dep was replaced with both of the new ones, but it only seems to use one or the other (in this case client-utils).

File: experimental/PropertyDDS/examples/property-inspector/package.json [hunk 2]
```diff
@@ -46,13 +46,14 @@
 		"@fluid-experimental/property-inspector-table": "workspace:~",
 		"@fluid-experimental/property-properties": "workspace:~",
 		"@fluid-experimental/property-proxy": "workspace:~",
+		"@fluid-internal/client-utils": "workspace:~",
 		"@fluidframework/aqueduct": "workspace:~",
-		"@fluidframework/common-utils": "^1.1.1",
```
Comment (alexvy86) [inline experimental/PropertyDDS/examples/property-inspector/package.json:50]: This one doesn't seem like it used the dep at all. I'd probably skip it altogether in this PR and just remove the dep on a separate one.

**Review by alexvy86 — COMMENTED** [review]
I left a few comments below. I totally undestand it's a huge pain to do the dep replacements by hand on each package, but especially for things under `packages/` (and honestly, a bit for everything else too) I would't want to leave unnecessary dependencies hanging around. Webpack would tree-shake them, but node apps don't get the same advantage, and it's easier to reason about the dependency graph if we don't have spurious ones declared in package.json.

File: experimental/PropertyDDS/examples/property-inspector/package.json (hunk 2 above)
Comment (tylerbutler) [reply experimental/PropertyDDS/examples/property-inspector/package.json:50]: While that would be more correct, possibly, it doesn't seem necessary. Part of the intent of this change is to remove all dependencies on common-utils, even if previously they weren't used, so to do it correctly I'd need to pull this change out into a separate change and merge it before I merge this.

File: examples/version-migration/schema-upgrade/package.json (hunk 1 above)
Comment (tylerbutler) [reply examples/version-migration/schema-upgrade/package.json]: Good catch. I found many of these were unused and fixed them in commit 727be7ae298707c466b3b36214f600ce42d6eeb4

To find them, I used this script to remove the deps if present:

```ts
const fs = require("fs");

const pkg = JSON.parse(fs.readFileSync("./package.json"));
if (pkg.dependencies) {
	delete pkg.dependencies["@fluidframework/common-utils"];
	if (
		Object.keys(pkg.dependencies).includes("@fluidframework/core-utils") &&
		Object.keys(pkg.dependencies).includes("@fluid-internal/client-utils")
	)
	delete pkg.dependencies["@fluidframework/core-utils"];
	delete pkg.dependencies["@fluid-internal/client-utils"];
	fs.writeFileSync("./package.json", JSON.stringify(pkg, undefined, "\t"));
} else {
	console.error("No dependency field");
	process.setMaxListeners(1);
}

process.exit(0);
```

I ran it against all packages in the client release group:

```shell
flub exec -g client -- "node /absolute/path/to/script.cjs"
```

Then I ran `pnpm i` and `pnpm build:compile`, fixed the errors, and repeated until done. Finally I merged the changes that survived.

File: experimental/PropertyDDS/examples/property-inspector/package.json (hunk 2 above)
Comment (alexvy86) [reply experimental/PropertyDDS/examples/property-inspector/package.json:50]: And now that the new deps aren't being added to it, I agree this is fine.

**Review by alexvy86 — APPROVED** [review]
🚀

**Pull request merged**